From eed7bac7d8e3246e955af32e8864440ffe11929e Mon Sep 17 00:00:00 2001
From: Arturo Reuschenbach Puncernau <reuschenbach@gmail.com>
Date: Wed, 30 Sep 2015 16:43:41 +0200
Subject: [PATCH] Fix endless loop after authenticate when the swift server is
 not reachable

This avoids a endless loop when getting an error from doTimeoutRequest an method call
---
 swift.go | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/github.com/ncw/swift/swift.go b/github.com/ncw/swift/swift.go
index 003e105..a75dc56 100644
--- a/github.com/ncw/swift/swift.go
+++ b/github.com/ncw/swift/swift.go
@@ -467,11 +467,11 @@ func (c *Connection) Call(targetUrl string, p RequestOpts) (resp *http.Response,
 		req.Header.Add("X-Auth-Token", authToken)
 		resp, err = c.doTimeoutRequest(timer, req)
 		if err != nil {
-			if p.Operation == "HEAD" || p.Operation == "GET" {
+			if (p.Operation == "HEAD" || p.Operation == "GET") && retries > 0 {
 				retries--
 				continue
 			}
-			return
+			return nil, nil, err
 		}
 		// Check to see if token has expired
 		if resp.StatusCode == 401 && retries > 0 {
